api/proto/teleport/crownjewel/v1/crownjewel.proto

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.crownjewel.v1;
+
+import "google/protobuf/wrappers.proto";
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/crownjewel/v1;crownjewelv1";
+
+// CrownJewel represents a Crown Jewel resource.
+// Crown Jewel is a resource that represents a set of resources that are
+// considered critical to the organization. Access Graph uses Crown Jewel to
+// generate audit events if access to a resource has changed.
+message CrownJewel {
+  // The kind of resource represented.
+  string kind = 1;
+  // Mandatory field for all resources. Not populated for this resource type.
+  string sub_kind = 2;
+  // The version of the resource being represented.
+  string version = 3;
+  // Common metadata that all resources share.
+  teleport.header.v1.Metadata metadata = 4;
+  // Spec is the crown jewel spec.
+  CrownJewelSpec spec = 5;
+}
+
+// CrownJewelSpec is the specification of a Crown Jewel.
+message CrownJewelSpec {
+  // TeleportMatchers is a list of teleport matchers.
+  repeated TeleportMatcher teleport_matchers = 1;
+  // AWSMatchers is a list of AWS matchers.
+  repeated AWSMatcher aws_matchers = 2;
+}
+
+// TeleportMatcher represents a matcher for Teleport resources.
+message TeleportMatcher {
+  // Name is the name of the resource. When the name is provided, it will match
+  // the resource with the same name.
+  string name = 1;
+  // Kind is the kind of the resource: ssh, k8s, db, etc
+  // Multiple kinds can be provided to match multiple kinds.
+  repeated string kinds = 2;
+  // Labels is a set of labels.
+  repeated TeleportLabel labels = 3;
```

Comment:
nit:  Can we reuse [teleprot.labelv.1Lables](https://github.com/gravitational/teleport/blob/9d7c59f9a40f6fa990106b1976ecb5925d1e58af/api/proto/teleport/label/v1/label.proto#L22) ? 
```suggestion
  repeated teleport.label.v1.Label labels = 3;
```